source/DSCResources/DSC_CMCollections/DSC_CMCollections.psm1

```diff
@@ -0,0 +1,685 @@
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $CollectionName,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('User','Device')]
+        [String]
+        $CollectionType,
+
+        [Parameter()]
+        [String]
+        $LimitingCollectionName,
+
+        [Parameter()]
+        [String]
+        $Comment,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $RefreshSchedule,
+
+        [Parameter()]
+        [ValidateSet('None','Periodic','Continuous','Both')]
+        [String]
+        $RefreshType,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance[]]
+        $QueryRules,
+
+        [Parameter()]
+        [String[]]
+        $ExcludeMembership,
+
+        [Parameter()]
+        [String[]]
+        $DirectMembership,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule
+    Set-Location -Path "$($SiteCode):\"
+
+    try
+    {
+        $state = Get-TargetResource -SiteCode $SiteCode -CollectionName $CollectionName -CollectionType $CollectionType
+        $cn = Get-CMCollection -Name $CollectionName
+
+        if ($Ensure -eq 'Present')
+        {
+            if ($null -eq $cn)
+            {
+                Write-Verbose -Message ($script:localizedData.CollectionAbsent -f $CollectionName)
+
+                $newCollection = @{
+                    Name                   = $CollectionName
+                    CollectionType         = $CollectionType
+                    LimitingCollectionName = $LimitingCollectionName
+                }
+
+                New-CMCollection @newCollection
+            }
+
+            $buildingParams = @{
+                Name = $CollectionName
+            }
+
+            $itemsForEval = @(
+                @{
+                    Name  = 'Comment'
+                    Value = $Comment
+                }
+                @{
+                    Name  = 'RefreshType'
+                    Value = $RefreshType
+                }
+            )
+
+            foreach ($item in $itemsForEval)
+            {
+                if ((-not [string]::IsNullOrEmpty($item.Value)) -and ($state[$item.Name] -ne $item.Value))
+                {
+                    Write-Verbose -Message ($script:localizedData.CollectionSetting -f $CollectionName, `
+                        $($item.name), $($item.Value), $($state[$item.Name]))
+
+                    $buildingParams += @{
+                        $($item.Name) = $($item.Value)
+                    }
+                }
+            }
+
+            if (-not [string]::IsNullOrEmpty($RefreshSchedule))
+            {
+                $newSchedule = @{
```

Comment:
Can you do, `New-CMSchedule @RefreshSchedule`
I know it't not a hash table but curious if it would translate.

Reply:
Not without a conversion would still need to happen